src/transformers/onnx/__main__.py

```diff
@@ -54,12 +52,13 @@ def main():
     # Check the modality of the inputs and instantiate the appropriate preprocessor
     # TODO(lewtun): Refactor this as a function if we need to check modalities elsewhere as well
     config = AutoConfig.from_pretrained(args.model)
-    if config.model_type in TOKENIZER_MAPPING_NAMES:
+    try:
         preprocessor = AutoTokenizer.from_pretrained(args.model)
-    elif config.model_type in FEATURE_EXTRACTOR_MAPPING_NAMES:
-        preprocessor = AutoFeatureExtractor.from_pretrained(args.model)
-    else:
-        raise ValueError(f"Unsupported model type: {config.model_type}")
+    except OSError:
+        try:
+            preprocessor = AutoFeatureExtractor.from_pretrained(args.model)
+        except OSError:
+            raise ValueError(f"Unsupported model type: {config.model_type}")
```

Comment:
I think these changes could break other ONNXConfig implementations that are relying on the past `if/elif/else` statement

Reply:
Yep, I wasn't sure about this. But I believe it should work for the other cases, given that they should return a tokenizer or feature extractor with the `from_pretrained` methods. I thought this would be a better solution than introducing a special case for Perceiver. Do you have an alternative suggestion?

Reply:
I noticed that we also have a check like this in the test cases: https://github.com/huggingface/transformers/blob/main/tests/onnx/test_onnx_v2.py#L263. There, we look at the `main_input_name` to decide whether we should instantiate a tokenizer or a feature extractor. The problem with this is that it won't work for models that can handle multiple modalities (such as Perceiver or `SpeechEncoderDecoderModel`-based ones) that have `main_input_name = "inputs"`.

I'm wondering whether the `try-catch` construct wouldn't be a well-suited way to address this problem, as it's also in the spirit of [easier to ask for forgiveness than permission](https://docs.python.org/3.5/glossary.html#term-eafp). I ran the slow tests, and they seem to work fine with it.

@lewtun I believe you wrote both of those checks. What do you think?

Reply:
Hey @lewtun! 

Yep, I gave it a try. Could you have a look? 😊 

The framework theoretically allows a tokenizer AND a feature extractor/processor to be returned, so it became a bit more complicated. I added a new parameter that currently won't ever be used. If you think it's overengineered, I can just remove it and raise an error in that case.